Sources/DulocCastle/ASN1/Asn1ObjectIdentifier.swift

```diff
@@ -0,0 +1,269 @@
+//
+//  Asn1ObjectIdentifier.swift
+//  
+//
+//  Created by Kevin Miller on 7/23/22.
+//
+
+import Foundation
+
+enum Asn1OidAccessError: Error {
+    case OidNotFoundErr
+    case OidInUseErr
+    case OidUuidNotAllowedErr
+    case UnexpectedOidErr(message: String, code: Int? = nil)
+}
+
+enum OidConversionError: Error {
+    case InvalidStringRepresentationGivenErr
+    case UnexpectedOidConversionError(message: String, code: Int? = nil)
+}
+
+// Asn.1 Object Identifier (This will be codable)
+struct Asn1Oid {
+    var obj: [UInt]
+    var str:  String
+    var name: String
+    
+    // Built In UUID will be 0x0000 - 0x1FFF
+    // Custom UUID can be 0x2000 - 0xFFFF
+    var uuid: UInt
+    
+    fileprivate init(
+        _ obj: [UInt],
+        _ str: String,
+        _ name: String,
+        _ uuid: UInt
+    ) {
+        self.obj = obj
+        self.str = str
+        self.name = name
+        self.uuid = uuid
+    }
+}
+
+
+/// The registry stores all of our OID objects within dictionaries for easy access using all of the identifiers.
+/// This also ensures that there is only one source of truth for the available OIDs
+/// For someone using the library, they will register their custom OID with the registry and
+/// it will be verifed. Upon verification, they can get it using any identifier.
+struct Asn1OidRegistry {
```

Comment:
This should be a regular struct with a static, shared singleton instance i.e. `Asn1OidRegistry.shared.something()`.